### ISSUE TITLE
Leaflet throwing "Cannot read property 'max' of undefined" when adding a populated group layer to a map

### ISSUE THREAD

**Issue by fplgusmao** [Leaflet/Leaflet#4199]
**Basically,** I'm making an app that, among other things, lets users insert `L.Circle`'s on a group layer on top of a map. Then, it saves the layers of that group layer – they are all Leaflet circles – in a variable located under an AngularJS Service. It then needs to load (add) them again into another group layer, which in turn will be added to a map. I add each of them to the group layer using `groupLayer.addLayer`. **When I add that group layer into a map, using `map.addLayer(groupLayer)`, Leaflet throws:**

`Cannot read property 'max' of undefined`

The call stack:

```
Cannot read property 'max' of undefined
    at L.Circle.L.Path.extend._checkIfEmpty (http://.../leaflet-src.js:5918:22)
    at L.Circle.L.Path.extend.getPathString (http://.../leaflet-src.js:5881:12)
    at L.Path.L.Path.extend._updatePath (http://.../leaflet-src.js:4682:18)
    at L.Path.L.Class.extend.onAdd (http://.../leaflet-src.js:4503:8)
    at L.Map.L.Class.extend._layerAdd (http://.../leaflet-src.js:2225:9)
    at L.Map.L.Class.extend.addLayer (http://.../leaflet-src.js:1683:9)
    at L.LayerGroup.L.Class.extend.eachLayer (http://.../leaflet-src.js:4322:11)
    at L.LayerGroup.L.Class.extend.onAdd (http://.../leaflet-src.js:4307:8)
    at L.Map.L.Class.extend._layerAdd (http://.../leaflet-src.js:2225:9)
    at L.Map.L.Class.extend.addLayer (http://.../leaflet-src.js:1683:9)
```

From what I understood, this is related to Leaflet not managing to render the circles... But why is this happening? **[I made a plunker](https://plnkr.co/eOrZXo)** (**updated**), trying to show what I am doing in my app. ~~However, **in the plunker everything is working fine**. It may still be useful by kinda showing what steps I'm taking, and what I'm trying to do. In my app things are structured a little differently, and I don't know if I can make a simple plunker like it. Do you think it can be related to how my code is structured? Anyway, if needed, I can try to explain how things work in detail.~~

[**UPDATE**] as @IvanSanchez said, it is a 0.7.x problem. I changed the leaflet version in the plunker and *now* the described problem is happening. You can now mess with it, and check the console error.

**Comment by IvanSanchez** [comment]
This is failing on some function calls specific to 0.7.x - so not a 1.0.0 issue at all.

@fplgusmao Unless you can provide a better plunker/jsfiddle/something, it's unlikely we'll spend time fixing this - we're focused on 1.0.0 issues. I can see the issue has something to do with the 0.7.x way of setting the `_pathViewport` on `L.Path`s. Make sure your SVG/Canvas renderer is instanced and init'ed beforehand, etc etc.

Or switch to 1.0.0-beta2.

**Comment by fplgusmao** [comment]
@IvanSanchez you're right, it really is a 0.7.x problem. I will try to change to 1.0.0 and report the result.

**Comment by IvanSanchez** [comment]
@fplgusmao Any updates? We'd like to close stale bugs.

**Comment by fplgusmao** [comment]
Nope. But I didn't dive too deep into the problem, though. I found that the 1.0 version was not compatible with some other plugins I was using, or with the way I was using them (e.g [Leaflet.FreeDraw](https://github.com/Wildhoney/Leaflet.FreeDraw) wasn't working), so the fastest way I found was simply to remove the feature of reloading a state and its markers, and stick to `Leaflet 0.7.7` for the moment. It worked for my situation, since this "marker reload" was not a crucial feature.

*To sum it up*: this bug is still happening, and I haven't found a way to properly work around it, or solve it.

Feel free to close the issue if you want to "move on" to `Leaflet 1.0`-only

**Comment by LookinGit** [comment]
Faced the same problem, solution:
```
var Map = L.map('#map', {});
Map._initPathRoot();
Map._updatePathViewport();
```
call when reinitialization.

**Comment by fplgusmao** [comment]
@LookinGit thanks a lot! it worked for me!
